src/modality-specific-files/physiological-recordings.md

```diff
@@ -72,122 +194,585 @@ A guide for using macros can be found at
+  "sub-01": {
+    "func": {
+      "sub-01_task-visualSearch_bold.json": "",
+      "sub-01_task-visualSearch_bold.nii.gz": "",
+      "sub-01_task-visualSearch_events.json": "",
+      "sub-01_task-visualSearch_events.tsv": "",
+      "sub-01_task-visualSearch_recording-eye1_physio.json": "",
+      "sub-01_task-visualSearch_recording-eye1_physio.tsv.gz": "",
+      "sub-01_task-visualSearch_recording-eye1_physioevents.json": "",
+      "sub-01_task-visualSearch_recording-eye1_physioevents.tsv.gz": ""
+    },
+  },
+}) }}
+
+The above example is extended to a binocular eye-tracker producing three signals
+(left and right eyes, plus a *cyclopean* recording), as follows:
+<!-- This block generates a file tree.
+A guide for using macros can be found at
+ https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md
+-->
+{{ MACROS___make_filetree_example({
+  "sub-01": {
+    "func": {
+      "sub-01_task-visualSearch_bold.json": "",
+      "sub-01_task-visualSearch_bold.nii.gz": "",
+      "sub-01_task-visualSearch_events.json": "",
+      "sub-01_task-visualSearch_events.tsv": "",
+      "sub-01_task-visualSearch_recording-eye1_physio.json": "",
+      "sub-01_task-visualSearch_recording-eye1_physio.tsv.gz": "",
+      "sub-01_task-visualSearch_recording-eye1_physioevents.json": "",
+      "sub-01_task-visualSearch_recording-eye1_physioevents.tsv.gz": "",
+      "sub-01_task-visualSearch_recording-eye2_physio.json": "",
+      "sub-01_task-visualSearch_recording-eye2_physio.tsv.gz": "",
+      "sub-01_task-visualSearch_recording-eye2_physioevents.json": "",
+      "sub-01_task-visualSearch_recording-eye2_physioevents.tsv.gz": "",
+      "sub-01_task-visualSearch_recording-eye3_physio.json": "",
+      "sub-01_task-visualSearch_recording-eye3_physio.tsv.gz": "",
+      "sub-01_task-visualSearch_recording-eye3_physioevents.json": "",
+      "sub-01_task-visualSearch_recording-eye3_physioevents.tsv.gz": ""
+    },
+  },
+}) }}
+
+Given the above example file structures, a corresponding
+`sub-01_task-visualSearch_recording-eye1_physio.json` sidecar
+could read:
+
+```JSON
+{
+    "DeviceSerialNumber": "17535483",
+    "Columns": ["timestamp", "x_coordinate", "y_coordinate", "pupil_size"],
+    "EnvironmentCoordinates": "top-left",
+    "Manufacturer": "SR-Research",
+    "ManufacturersModelName": "EYELINK II CL v4.56 Aug 18 2010",
+    "RecordedEye": "right",
+    "SampleCoordinateSystem": "gaze-on-screen",
+    "SampleCoordinateUnits": "pixel",
+    "SamplingFrequency": 1000,
+    "SoftwareVersion": "SREB1.10.1630 WIN32 LID:F2AE011 Mod:2017.04.21 15:19 CEST",
+    "ScreenAOIDefinition": [
+        "square",
+        [100, 150, 300, 350]
+    ],
+    "pupil_size": {
+        "Description": "Pupil area of the recorded eye as calculated by the eye-tracker in arbitrary units (see EyeLink's documentation for conversion).",
+        "Units": "a.u."
+    }
+}
+```
+
+Content of `sub-01_task-VisualSearch_events.json`:
```

Comment:
Is this really the _events.json file or is it the _physioevents.json?

Reply:
This is correct, it annotates the events.tsv file, as opposed to the new _physioevents.json.

Reply:
So basically in the _physioevents.json file I only have the column names plus their description? And do I need a _events.json file if I do not have an _events.tsv? Because Martins dataset on Open Neuro does not have it.

Reply:
Okay, gotcha -- you've actually hit an edge case we do not currently cover:

- We say that some metadata belonging in the `_events.json` file becomes mandatory if eye tracking is present
- However, BIDS allows `task-rest` to dismiss the events.tsv file. This BEP could make it mandatory for rest, even if the `events.tsv` file is just empty (only the header row).

WDYT @effigies @Remi-Gau ?

Reply:
(dataset: https://doi.org/10.18112/openneuro.ds004158.v2.0.2)

Reply:
Or include it in the `physioevents.json`? (if this does not disturb too much other stuff) Because that would be easy to include in the converter as we are already asking users to put some metadata manually in the` metadata.yml ` which is read in by our code. The current code actually does that because I understood it wrong, cf my initial question...

Reply:
Not sure that is a good idea. If I were to move the metadata from events.json, I would think it'd be better in the physio.json file rather than physioevents.json.

These metadata are about stimulus presentation, so they are likely relevant beyond eye tracking exclusively. Perhaps a reasonable alternative would be to update the specs so these metadata are encoded within stim.json files instead of the events.json files. Either way, they need to allow a json be defined without the corresponding .tsv[.gz] file, or allow empty tables (okay for .tsv but dangerous for .tsv.gz).

Reply:
Ok, I'm of course fine with the solution that fits best to BIDS in general but I agree that we should keep this information somehow. 

> allow a json be defined without the corresponding .tsv[.gz] file

would probably the best solution for this particular BEP as we agreed on .tsv.gz